frontera/contrib/backends/sqlalchemy/components.py

```diff
@@ -123,18 +123,18 @@ def frontier_stop(self):
 
     @retry_and_rollback
     def fetch(self, fingerprints):
-        to_fetch = [f for f in fingerprints if f not in self._cache]
+        to_fetch = [to_native_str(f) for f in fingerprints if f not in self._cache]
```

Comment:
Since `f` is iterated over `fingerprints`(which contains fingerprints as bytes), the search is carried out using f as bytes. The resulting list will have these fingerprints as native_str. `to_fetch` is then used to query the DB where we want them to be as native strings.